eclipse-collections/src/main/java/org/eclipse/collections/impl/list/fixed/ArrayAdapter.java

```diff
@@ -180,6 +182,17 @@ public T set(int index, T element)
         return new ArrayAdapter<>(this.items.clone());
     }
 
+    /**
+     * Overrides default method from List.
+     *
+     * @since 10.0
+     */
+    @Override
+    public void sort(Comparator<? super T> comparator)
+    {
+        Arrays.sort(this.items, 0, this.size(), comparator);
```

Comment:
I was expecting sort() would delegate to sortThis(). I was expecting an attempt to add the implementation to MutableList. It's possible that would lead to diamond inheritance problems and need to get pushed down, but it probably still makes sense to delegate from one to the other. Thoughts? In this case, the implementation of sort() and sortThis() in ArrayAdapter are different and probably shouldn't be. Either they should both get the optimization for small sizes, or neither.

Reply:
Alternately, sortThis could delegate to sort(). Either way, it should appear that there are no new implementations of sorting.

Reply:
I tried this initially, but ran into issues with subList().sort().  This method caused StackOverflows errors.

```
protected void defaultSort(Comparator<? super T> comparator)
{
    Collections.sort(this, comparator);
}
```
I just tried this approach again, and remembered why I opted to go with explicit implementations.  I'll see if there is something to do that will fix this.  Help appreciated.

Reply:
I'm going to try and implement sort and delegate to sort from sortThis.  I think this conceptually makes the most sense.